cpp/include/cuspatial/experimental/detail/point_linestring_distance.cuh

```diff
@@ -120,11 +177,12 @@ void pairwise_point_linestring_distance(Cart2dItA points_first,
   static_assert(detail::is_same<vec_2d<T>,
                                 detail::iterator_value_type<Cart2dItA>,
                                 detail::iterator_value_type<Cart2dItB>>(),
-                "Inputs must be cuspatial::vec_2d<T>");
+                "Inputs must be cuspatial::vec_2d");
 
-  auto const num_pairs = thrust::distance(points_first, points_last);
+  auto const num_pairs =
+    thrust::distance(point_geometry_offset_first, point_geometry_offset_last) - 1;
 
-  if (num_pairs == 0) { return; }
+  if (num_pairs == 0) { return distances_first; }
```

Comment:
Need to check that the `num_pairs` is also equal to the number of linestrings in `linestring_geometry_offset_first`, or the number of `linestring_part_offsets` if `linestring_geometry_offset_first` is `nullptr`?

I'm not sure from having read yet but I think I saw that either the algorithm treats the multilinestring and linestring situations as different cases? What about always passing in the multilinestring offsets buffer, creating it if one isn't included by the caller, and making it as simple as `[0, 1]` if only a single point/linestring is passed in?

Reply:
The current algorithm *is* basically what you mentions above :) Except users can use a counting iterator to avoid materializing the sequence. `linestring_geometry_offset_first` is assumed to never be null.